crypto/store/store_lib.c

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright 2001-2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <openssl/crypto.h>
+#include <openssl/err.h>
+#include <openssl/store.h>
+#include "internal/thread_once.h"
+#include "store_local.h"
+
+struct store_ctx_st {
+    const STORE_LOADER *loader;
+    STORE_LOADER_CTX *loader_ctx;
+    const UI_METHOD *ui_method;
+    void *ui_data;
+    STORE_post_process_info_fn post_process;
+    void *post_process_data;
+};
+
+STORE_CTX *STORE_open(const char *uri, const UI_METHOD *ui_method,
+                      void *ui_data, STORE_post_process_info_fn post_process,
+                      void *post_process_data)
+{
+    const STORE_LOADER *loader;
+    STORE_LOADER_CTX *loader_ctx = NULL;
+    STORE_CTX *ctx = NULL;
+    char scheme_copy[256], *p;
+
+    OPENSSL_strlcpy(scheme_copy, uri, sizeof(scheme_copy));
+    if ((p = strchr(scheme_copy, ':')) != NULL) {
+        *p = '\0';
+        p = scheme_copy;
+    } else {
+        p = "file";
```

Comment:
> That would make line 39 rather difficult, wouldn't it?

You are reusing `p` for two different purposes. One as an index into your local scheme_copy (which is non const), and again for a different purpose to point to the scheme name currently in use. I'd argue that those should be two different variables (one declared const and one not).